Source/Core/Core/IOS/DolphinDevice.cpp

```diff
@@ -123,6 +129,37 @@ IPCCommandResult SetSpeedLimit(const IOCtlVRequest& request)
   return DolphinDevice::GetDefaultReply(IPC_SUCCESS);
 }
 
+IPCCommandResult GetRealProductcode(const IOCtlVRequest& request)
+{
+  if (!request.HasNumberOfValidVectors(0, 1))
+  {
+    return DolphinDevice::GetDefaultReply(IPC_EINVAL);
+  }
+
+  const std::string backup_file_path = File::GetUserPath(D_BACKUP_IDX) + DIR_SEP + WII_SETTING;
+
+  File::IOFile file(backup_file_path, "rb");
+  Common::SettingsHandler::Buffer data;
+
+  if (file && file.ReadBytes(data.data(), data.size()))
```

Comment:
I'd invert the condition here and return early on failure so that the rest of the function can be unindented.